### DISCUSSION TITLE
How to screenshot a particular element?

### DISCUSSION THREAD

**Discussion by lofti198** [cefsharp/CefSharp#4607] [Q&A]
I have and xpath (or another selector type) of element and I need to make a screenshot of this element after page is rendered and save it to any format I can use later to manipulate with the screenshot. What is the general approach for this?

**Comment by amaitland** [comment]
Screenshots are primarily captured using [Page.captureScreenshot](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-captureScreenshot) (DevTools protocol). 

You should be able to use the `clip` param to only capture a specific region.

1. Directly using CDP
- https://github.com/cefsharp/CefSharp/blob/2cb81aa943ed3a9787709bfa1764e91abf2bfbe3/CefSharp.Example/DevTools/DevToolsExtensions.cs#L20

2. Using CefSharp.Dom
Alternatively you can probably use https://github.com/cefsharp/CefSharp.Dom which should support capturing a specific DOM element.

- https://github.com/cefsharp/CefSharp.Dom/blob/main/lib/PuppeteerSharp/ElementHandle.cs#L127
